tuning/data/data_handlers.py

```diff
@@ -247,16 +252,80 @@ def apply_tokenizer_chat_template(
         Formatted HF Dataset element by formatting dataset with tokenizer's chat template
         Saves the result to dataset_text_field argument.
     """
+    chat_template_key = kwargs.get("chat_template_key", None)
```

Comment:
could we rename this to something else? possibly `chat_data_key` 

also if this is an important field I'd suggest making this a natural argument instead of hiding in kwargs.